### PR TITLE
Story#509 10

### PR DESCRIPTION
**Pull request by srikanthgudipudi**

### CODE REVIEW

File: src/styles.css [hunk 1]
```diff
@@ -2557,5 +2554,16 @@ margin: 0 2px;
   content:" *";
   /* color: red; */
 }
+.mat-tab-header-pagination-controls-enabled .mat-tab-header-pagination{
+  display: none !important;
+}
+.mat-icon {
```
Comment (kiranharidas187) [inline src/styles.css]: Please don't override the library classes unless it's a common use case, use a class instead

File: src/app/header/header.component.ts [hunk 2]
```diff
@@ -28,4 +32,11 @@ export class HeaderComponent implements OnInit {
     document.getElementById("ham-line").classList.toggle("active")
   }
 
+  /**
+   * This is used to logout from the application
+   */
+  logout(){
+    this.KeycloakService.getLogout();
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.ts]: Calling service from here means?

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -1,36 +1,27 @@
...
   },
-  {
-    path: '',
-    redirectTo: 'user',
-    pathMatch: 'full'
-  }
-
-
+  { path: 'home', component: CommingSoonComponent },
+  { path: 'unauthor', component: UnauthorizedComponent },
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts]: Can you please suggest the name here

File: src/app/app.component.ts [hunk 4]
```diff
@@ -9,16 +11,19 @@ import { Router } from '@angular/router';
...
+    this.userdetails = this.KeycloakService.getCurrentUserDetails();
+    localStorage.setItem('userName', this.userdetails.name);
+    // localStorage.setItem('roleInfo', this.userdetails.resource_access.account.roles);
   }
 
 
   // Initial loading
   ngOnInit() {
-    this.router.navigate(['/user/user']);
+    this.router.navigate(['/users']);
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/app.component.ts]: If navigation is not given in the appcomponent, then when ever logout and login then it is loading previous path that's why i maintained router here

File: src/app/header/header.component.html [hunk 5]
```diff
@@ -1,231 +1,86 @@
...
+      </div>
+      <div class="col-md-4">
+        <a [routerLink]="['/users']" [routerLinkActive]="['active']"
+          [ngClass]="{'active': singleUser.isActive || multiusersActive.isActive}">
+          Create Users</a>
+        <a [routerLink]="['/users']" routerLinkActive #usersActive="routerLinkActive" style="display: none">
+          Users</a>
+      </div>
+      <div class="col-md-5">
+        <a [routerLink]="['/multi-users']" routerLinkActive #multiusersActive="routerLinkActive" style="display: none">
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.html]: Changed

File: src/app/header/header.component.html [hunk 6]
```diff
@@ -1,231 +1,86 @@
...
+          Assign Users</a>
+        <a [routerLink]="['/unauthor']" [routerLinkActive]="['active']">
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.html]: If he is unauthorised to any page , then i will redirect to this page

File: src/app/modules/users/single-user/single-user.component.html [hunk 7]
```diff
@@ -8,13 +8,13 @@ <h3 class="titlehead">User Info</h3>
...
-                        </span>
-                        <h3 class="ftitle">User Creation</h3>
+                <span class="dashboard-icon">
+                    <mat-icon class="roundimg">person</mat-icon>
+                </span>
+                <h3 class="ftitle">User Creation</h3>
             </div>
-            <div class="col-md-3">
-               
+            <div class="col-md-2">
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/modules/users/single-user/single-user.component.html]: I need space between them, that's why i used empty div

File: src/app/app-routing.module.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/app-routing.module.ts]: unauthorized

File: src/app/app.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/app.component.ts]: Please put this navigation in logout function inside keycloakservice

File: src/app/app.module.ts [hunk 8]
```diff
@@ -6,12 +6,11 @@ import { CoreModule, TranslateService, SharedModule } from 'shikshalokam';
...
 import { AppComponent } from './app.component';
 import { HeaderComponent } from './header/header.component';
-import { SidenavComponent } from './sidenav/sidenav.component';
-import { LayoutComponent } from './layout/layout.component';
 import { LoginComponent } from './login/login.component';
 import { JwtHelperService, JwtModule } from '@auth0/angular-jwt';
-import { KeyCloakService } from './modules/keycloak/keycloak.service';
-
+import { AdminSharedModule, KeyCloakService,
+  MatButtonModule, MatTabsModule, MatToolbarModule, MatIconModule, MatMenuModule } from './modules/admin-shared';
```
Comment (kiranharidas187) [inline src/app/app.module.ts]: For material modules please follow the same thing used in 
https://github.com/shikshalokam/sl-assessments-portal/blob/master/src/app/shared/shared.module.ts

File: src/app/header/header.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/header/header.component.html]: Why is this given as display none?

File: src/app/header/header.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/header/header.component.html]: This redirection should be given in route guards.

File: src/app/header/header.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/header/header.component.ts]: just rename the this.KeycloakService.getLogout(); to this.KeycloakService.logout();

Rest everything is fine

File: src/app/modules/admin-shared/index.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export * from './comming-soon/comming-soon.component';
+export * from './unauthorized/unauthorized.component';
+export * from './keycloak/keycloak.service';
+export * from './admin-shared.module';
+export * from '@angular/material';
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/index.ts]: no need to export @angular/material

File: angular.json [hunk 10]
```diff
@@ -23,17 +27,15 @@
...
+              "src/styles.css"
             ],
             "scripts": [
               "./node_modules/jquery/dist/jquery.js",
-              "./node_modules/popper.js/dist/umd/popper.min.js", 
+              "./node_modules/popper.js/dist/umd/popper.min.js",
               "./node_modules/bootstrap/dist/js/bootstrap.js",
-              "./node_modules/keycloak-js/dist/keycloak.min.js"
+              "./node_modules/keycloak-js/dist/keycloak.min.js",
+              "./node_modules/hammerjs/hammer.min.js"
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply angular.json]: while clicking on button , action is not happening properly, it is showing warnings like "long keypress", then i added this to identify the events

File: src/app/header/header.component.html [hunk 11]
```diff
@@ -1,231 +1,86 @@
...
-         <div class="col-md-5">
-            <a [routerLink]="['']" class="active">
-             Assign Users</a>
-         </div>
+  <div class="col-md-4 col-sm-8" style="padding-top: 10px;">
+    <div class="row font-weight-bold">
+      <div class="col-md-3">
+        <a [routerLink]="['/home']" [routerLinkActive]="['active']">
+          Home</a>
+        <a [routerLink]="['/single-user']" routerLinkActive #singleUser="routerLinkActive" style="display: none">
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.html]: Changed

File: src/app/header/header.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.html]: Modified

File: angular.json [hunk 12]
```diff
@@ -23,17 +27,15 @@
...
-              "node_modules/primeicons/primeicons.css"
+              "src/styles.css"
             ],
             "scripts": [
               "./node_modules/jquery/dist/jquery.js",
-              "./node_modules/popper.js/dist/umd/popper.min.js", 
+              "./node_modules/popper.js/dist/umd/popper.min.js",
               "./node_modules/bootstrap/dist/js/bootstrap.js",
-              "./node_modules/keycloak-js/dist/keycloak.min.js"
+              "./node_modules/keycloak-js/dist/keycloak.min.js",
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply angular.json]: Removed

File: src/app/app-routing.module.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts]: changed

File: src/app/app.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/app.component.ts]: Changed

File: src/app/header/header.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.html]: Just normal link,later if actual component comes then i will change

File: src/app/header/header.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/header/header.component.ts]: changed

File: src/app/app.module.ts [hunk 13]
```diff
@@ -1,16 +1,19 @@
...
 import { HeaderComponent } from './header/header.component';
-import { SidenavComponent } from './sidenav/sidenav.component';
-import { LayoutComponent } from './layout/layout.component';
 import { LoginComponent } from './login/login.component';
-import { JwtHelperService, JwtModule } from '@auth0/angular-jwt';
-import { KeyCloakService } from './modules/keycloak/keycloak.service';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import {
+  AdminSharedModule,
+  MatButtonModule, MatTabsModule, MatToolbarModule, MatIconModule, MatMenuModule
```
Comment (kiranharidas187) [inline src/app/app.module.ts]: Still, the material modules are imported here.

File: src/app/header/header.component.scss [hunk 14]
```diff
@@ -0,0 +1,17 @@
+  .space {
+    padding: 15px;
+}
+.atext{
+  text-align: right;
+}
+.mat-tab-header-pagination-controls-enabled .mat-tab-header-pagination{
```
Comment (kiranharidas187) [inline src/app/header/header.component.scss]: still, material classes edited directly. Please use a class to overirde

File: src/app/modules/admin-core/services/auth-service/auth.service.ts [hunk 15]
```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
```
Comment (kiranharidas187) [inline src/app/modules/admin-core/services/auth-service/auth.service.ts]: If we are not using this please remove this service

File: src/app/modules/admin-shared/index.ts [hunk 16]
```diff
@@ -0,0 +1,4 @@
+export * from './comming-soon/comming-soon.component';
+export * from './unauthorized/unauthorized.component';
+export * from './admin-shared.module';
+export * from '@angular/material'
```
Comment (kiranharidas187) [inline src/app/modules/admin-shared/index.ts]: Don't export @angular/material, its already exported by the angular material library

File: src/app/modules/users/multi-users/multi-users.component.scss [hunk 17]
```diff
@@ -20,14 +20,11 @@
 }
 
 .carddata{
-    width: 360px;
-    height:360px;
-    background-color: lightblue !important;
+    width: 468px;
+    height:325px;
+    background: #D6D6D6;
```
Comment (kiranharidas187) [inline src/app/modules/users/multi-users/multi-users.component.scss]: all the colors should be a variable in variable.scss

File: src/app/modules/users/single-user/single-user.component.scss [hunk 18]
```diff
@@ -5,13 +5,14 @@
     font-weight: 800;
     color: #a63936;
```
Comment (kiranharidas187) [inline src/app/modules/users/single-user/single-user.component.scss]: All the colors should be variables

File: src/app/modules/users/users.module.ts [hunk 19]
```diff
@@ -3,40 +3,29 @@ import { CommonModule } from '@angular/common';
...
 import { UsersComponent } from './users.component';
 import { MultiUsersComponent } from './multi-users/multi-users.component';
-import {FormsModule,ReactiveFormsModule} from '@angular/forms';
-import {TableModule} from 'primeng/table';
-import {TabViewModule} from 'primeng/tabview';
-import {CardModule} from 'primeng/card';
-import {ToastModule} from 'primeng/toast';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { SingleUserComponent } from './single-user/single-user.component';
+import { AdminSharedModule, MatCardModule, MatButtonModule, MatIconModule, MatTabsModule } from '../admin-shared';
```
Comment (kiranharidas187) [inline src/app/modules/users/users.module.ts]: Import the material modules in the shared module only

File: src/styles.css [hunk 20]
```diff
@@ -558,6 +551,10 @@ img{max-width: 100%;}
     width: fit-content;
     border-radius: 3px;
     margin: 0 auto;
+    margin-top: 10%;
```
Comment (kiranharidas187) [inline src/styles.css]: use margin :top right bottom left;

File: src/app/modules/admin-shared/index.ts (hunk 16 above)
Comment (srikanthgudipudi) [reply src/app/modules/admin-shared/index.ts]: changed

File: angular.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply angular.json]: Please remove this line

File: angular.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply angular.json]: Please remove this line

File: src/app/app.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/app.component.ts]: Please remove this line

File: src/app/header/header.component.scss [hunk 21]
```diff
@@ -0,0 +1,14 @@
+  .space {
+    padding: 15px;
+}
+.atext{
+  text-align: right;
+}
+
+.active {
+  border-bottom: 3px solid #a63936;
```
Comment (kiranharidas187) [inline src/app/header/header.component.scss]: colors should be in one variable scss file

File: src/app/modules/users/single-user/single-user.component.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (kiranharidas187) [reply src/app/modules/users/single-user/single-user.component.html]: Please make use of padding css property

File: src/styles.scss [hunk 22]
```diff
@@ -2557,5 +2553,16 @@ margin: 0 2px;
   content:" *";
   /* color: red; */
 }
+.mat-tab-header-pagination-controls-enabled .mat-tab-header-pagination{
```
Comment (kiranharidas187) [inline src/styles.scss]: Still editing the material class. Please use a custom class to override

File: src/styles.scss [hunk 23]
```diff
@@ -2557,5 +2553,16 @@ margin: 0 2px;
...
+  display: none !important;
+}
+.mat-icon {
```
Comment (kiranharidas187) [inline src/styles.scss]: Still editing the material class. Please use a custom class to override

File: src/app/modules/users/users.module.ts [hunk 24]
```diff
@@ -3,40 +3,28 @@ import { CommonModule } from '@angular/common';
...
-import {TabViewModule} from 'primeng/tabview';
-import {CardModule} from 'primeng/card';
-import {ToastModule} from 'primeng/toast';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { SingleUserComponent } from './single-user/single-user.component';
+import { AdminSharedModule } from '../admin-shared';
+
+
 
 const routes: Routes = [
```
Comment (kiranharidas187) [inline src/app/modules/users/users.module.ts]: Routing module should be separate..

**Pull request merged**